src/storage.js

```diff
@@ -416,6 +416,72 @@ BadgerStorage.prototype = {
     setTimeout(function(){
       _syncStorage(self);
     }, 0);
+  },
+
+  /**
+   * Merge the object into the existing map
+   *
+   * @param {Object} list The object containing storage map data to merge
+   */
+  mergeObject: function(list) {
```

Comment:
See if this can be made easier and simpler by using underscore. 
Underscore will have to be manually included in options.html